packages/core/content-releases/admin/src/components/ReleasesPanel.tsx

```diff
@@ -0,0 +1,141 @@
+import { useRBAC } from '@strapi/admin/strapi-admin';
+import { unstable_useDocumentLayout as useDocumentLayout } from '@strapi/content-manager/strapi-admin';
+import { Box, Flex, Typography } from '@strapi/design-system';
+import { useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+
+import { PERMISSIONS } from '../constants';
+import { useGetReleasesForEntryQuery } from '../services/release';
+import { getTimezoneOffset } from '../utils/time';
+
+import { ReleaseActionMenu } from './ReleaseActionMenu';
+
+import type { PanelComponent, PanelComponentProps } from '@strapi/content-manager/strapi-admin';
+
+const Panel: PanelComponent = ({ model, documentId, collectionType }: PanelComponentProps) => {
+  const { locale } = useParams<{
+    locale: string;
+  }>();
+
+  const {
+    edit: { options },
+  } = useDocumentLayout(model);
+  const { formatMessage, formatDate, formatTime } = useIntl();
+
+  const { allowedActions } = useRBAC(PERMISSIONS);
+  const { canRead, canDeleteAction } = allowedActions;
+
+  const response = useGetReleasesForEntryQuery({
+    contentTypeUid: model,
+    documentId,
+    locale,
+    hasEntryAttached: true,
+  });
+  const releases = response.data?.data;
+
+  const getReleaseColorVariant = (
+    actionType: 'publish' | 'unpublish',
+    shade: '100' | '200' | '600'
+  ) => {
+    if (actionType === 'unpublish') {
+      return `secondary${shade}`;
+    }
+
+    return `success${shade}`;
+  };
+
+  // Project is not EE or contentType does not have draftAndPublish enabled
+  if (!window.strapi.isEE || !options?.draftAndPublish || !canRead) {
+    return null;
+  }
+
+  if (collectionType === 'collection-types' && (!documentId || documentId === 'create')) {
+    return null;
+  }
+
+  return {
+    title: formatMessage({
+      id: 'content-releases.plugin.name',
+      defaultMessage: 'Releases',
+    }),
+    content: (
+      <Flex direction="column" alignItems="stretch" gap={3} width="100%">
+        {releases?.map((release: any) => (
```

Comment:
I think something is off here. The type comes through so we shouldn't need `any`. Removing it then gives an error and looking at the type, `actions` is not an array:

https://github.com/strapi/strapi/blob/c9e680cf69a00f2ccbe5984c3618e5cb7f8b9da8/packages/core/content-releases/shared/contracts/releases.ts#L24-L27

Reply:
Thank you, I was using the wrong type for this query response